### PR TITLE
Readme revision and example code refactorings

### PR DESCRIPTION
**Pull request by DavidVujic**
## Description

Simplified README file. Changed the recommended usage to a `promise` based client. 
Refactored the code in the `examples` folder to promise based.

## Motivation and Context

To simplify the documentation, and remove text that is no longer relevant.

Issue #166 

## How Has This Been Tested?

Travis CI Build and manual run of the examples code: `node examples/index.js`

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] I have read the [code of conduct](https://github.com/yfinkelstein/node-zookeeper/blob/master/CODE-OF-CONDUCT.md).
- [x] I have read the [contributing guide](https://github.com/yfinkelstein/node-zookeeper/blob/master/CONTRIBUTING.md).
- [x] I have updated the documentation accordingly (if applicable).

### CODE REVIEW

**Comment by DavidVujic** [comment]
TODO:
* modify the example ACL code in the README.
* write and run ACL code in the examples folder

**Comment by jbienkowski311** [comment]
README is not entirely true or at least it is confusing. You are using Zookeeper client and Zookeeper.Promise client interchangeably. What we could do, without changing the API, is to make callback optional. If the callback is provided then we use the method as we do right now. Otherwise we use async method and return promise.

**Comment by DavidVujic** [comment]
Oh, I thought I'm using the promise version in all examples?

When using `const ZooKeeper = require('zookeeper').Promise;` I am thinking that it is an active choice to use async/await/promises.

Maybe I'm missing something?

**Comment by DavidVujic** [comment]
Maybe the `connect` function need a Promise based version?

**Comment by DavidVujic** [comment]
Or, a simpler solution: use the `init` function instead. The `connect` function uses that one.

File: README.md [hunk 1]
```diff
@@ -1,97 +1,146 @@
...
+const client = createClient();
+```
+The client is ready when connected to a ZooKeeper server:
+```javascript
+client.init(config);
 
-    console.log(`zk session established, id=${client.client_id}`);
-    client.a_create('/node.js1', 'some value', ZooKeeper.ZOO_SEQUENCE | ZooKeeper.ZOO_EPHEMERAL, onCreate.bind(null, client));
-}
+await client.on_connected();
```
Comment (DavidVujic) [inline README.md]: I'm having second thoughts about this one. The function will probably cause confusion and likely bugs too. I'll remove it from the docs, and mark the function as deprecated.

Motivation: the `on_connected` function could potentially miss the connect event, if executed after init. If done before, `await` cannot be used.

**Comment by DavidVujic** [comment]
Thanks @jbienkowski311! I'll merge the PR.

**Pull request merged**